src/test/java/com/bugsnag/AppenderTest.java

```diff
@@ -0,0 +1,198 @@
+package com.bugsnag;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import com.bugsnag.delivery.Delivery;
+import com.bugsnag.delivery.HttpDelivery;
+import com.bugsnag.serialization.Serializer;
+
+import org.apache.log4j.Logger;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+
+/**
+ * Test for the Bugsnag Appender
+ * NOTE: Not called BugsnagAppenderTest because that throws away errors to prevent cycles
+ */
+public class AppenderTest {
```

Comment:
I agree with @fractalwrench, this is a little light on test scenarios.

Reply:
added more tests